### PR TITLE
targets crud operations

### PR DESCRIPTION
**Pull request by NiGiInRo**
#Sumary
- "Targets" basic Crud operations, association with Topics model. As same as Topics only aviable "index" and "create" routes, and all attributes validations as true.

#Card
- https://trello.com/c/chJA27Hh/46-rslc-2-as-a-user-i-should-be-able-to-create-a-new-target-so-i-can-find-people-with-similar-interests-in-a-specific-area

### CODE REVIEW

File: app/models/target.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Target < ApplicationRecord
+  belongs_to :topic
+  validates :tittle, presence: true
```
Comment (diego-suarez) [inline app/models/target.rb]:
```suggestion
  validates :title, presence: true
```

File: app/controllers/targets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    @target.destroy
+  end
+
+  private
+    def set_target
+      @target = Target.find(params[:id])
+    end
+
+    def target_params
+      params.require(:target).permit(:topic_id, :tittle, :radius, :lat, :lng)
```
Comment (diego-suarez) [inline app/controllers/targets_controller.rb]:
```suggestion
      params.require(:target).permit(:topic_id, :title, :radius, :lat, :lng)
```

File: app/models/target.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  validates :radius, presence: true
+  validates :lat, presence: true
+  validates :lng, presence: true
+  validates :topic_id, presence: true
```
Comment (diego-suarez) [inline app/models/target.rb]: you can simplify on only one line:

`validates :title, :radius, :lat, :lng, :topic_id, presence: true`

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  resources :targets, only: [:create, :index]
```
Comment (diego-suarez) [inline config/routes.rb:2]: how about other ones (show, destroy, update)? or are not part of this card?

File: db/migrate/20230511160143_create_targets.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+class CreateTargets < ActiveRecord::Migration[7.0]
+  def change
+    create_table :targets do |t|
+      t.references :topic, null: false, foreign_key: true
+      t.string :tittle
+      t.float :radius
+      t.float :lat
+      t.float :lng
```
Comment (diego-suarez) [inline db/migrate/20230511160143_create_targets.rb]: remember to add database validation too here when checking for `presence: true` on the model:
`null: false`

**Review by diego-suarez — COMMENTED** [review]
well done! left some comments

File: app/models/target.rb (hunk 3 above)
Comment (NiGiInRo) [reply app/models/target.rb]: wow, looks more clean. thanks!

File: app/models/target.rb (hunk 1 above)
Comment (NiGiInRo) [reply app/models/target.rb]: You right! a little mistake hehe. i'll fix it!

File: app/controllers/targets_controller.rb (hunk 2 above)
Comment (NiGiInRo) [reply app/controllers/targets_controller.rb]: same here

File: config/routes.rb (hunk 4 above)
Comment (NiGiInRo) [reply config/routes.rb:2]: on the card it say: "As a user I should be able to create a new target so I can find people with similar interests in a specific area", doesn't say much hehe. I just interpret that only a GET and POST is necessary

File: db/migrate/20230511160143_create_targets.rb (hunk 5 above)
Comment (NiGiInRo) [reply db/migrate/20230511160143_create_targets.rb]: you right! thanks

File: app/controllers/targets_controller.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def destroy
+    @target.destroy
+  end
+
+  private
+    def set_target
+      @target = Target.find(params[:id])
+    end
```
Comment (enzofab91) [inline app/controllers/targets_controller.rb:39]: You can memoize the variable, see https://www.honeybadger.io/blog/ruby-rails-memoization/

```suggestion
    def target
      @target ||= Target.find(params[:id])
    end
```

File: db/migrate/20230511160143_create_targets.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+class CreateTargets < ActiveRecord::Migration[7.0]
+  def change
+    create_table :targets do |t|
+      t.references :topic, null: false, foreign_key: true
+      t.string :tittle
```
Comment (enzofab91) [inline db/migrate/20230511160143_create_targets.rb]:
```suggestion
      t.string :title
```

File: app/models/target.rb (hunk 3 above)
Comment (enzofab91) [reply app/models/target.rb]: What if you also validate also that `radius` is greatear than 0?

**Review by enzofab91 — COMMENTED** [review]
Nice work! Left some comments 🙌

**Review by enzofab91 — APPROVED** [review]
Good work!

**Review by diego-suarez — APPROVED** [review]
🤘

**Pull request merged**